src/rosdistro/dependency_walker.py

```diff
@@ -48,7 +49,9 @@ def _get_package(self, pkg_name):
             pkg_xml = self._distribution_instance.get_release_package_xml(pkg_name)
             try:
                 pkg = parse_package_string(pkg_xml)
-            except InvalidPackage as e:
+                if self._evaluate_condition_context:
```

Comment:
Please check for `is not None` explicitly.

Same below.

Reply:
Done, thanks!